fluent/client/ws_client.go

```diff
@@ -117,10 +145,19 @@ func (c *WSClient) Connect() error {
 		return err
 	}
 
-	c.Session = &WSSession{
-		ServerAddress: c.ServerAddress,
-		Connection:    connection,
-	}
+	c.Session = c.ConnectionFactory.NewSession(connection)
+	waitChan := make(chan struct{}, 1)
+
+	go func() {
+		waitChan <- struct{}{}
+
+		if err := c.Session.Connection.Listen(); err != nil {
```

Comment:
The client may not be explicitly handling reads, so they may not know that the connection has been terminated. Store any connection errors here so they can be bubbled to the user next time they call `SendMessage`